packages/next/src/client/components/react-dev-overlay/internal/hooks/use-on-click-outside.ts

```diff
@@ -20,7 +20,9 @@ export function useOnClickOutside(
 
     const root = el.getRootNode()
     root.addEventListener('mousedown', listener as EventListener)
-    root.addEventListener('touchstart', listener as EventListener)
+    root.addEventListener('touchstart', listener as EventListener, {
+      passive: false,
+    })
     return function () {
       root.removeEventListener('mousedown', listener as EventListener)
       root.removeEventListener('touchstart', listener as EventListener)
```

Comment:
I think you need to match options otherwise it doesn't get removed properly.

Reply:
You don't need that option in removeEventListener, only `options.capture` matter. Details: https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/removeEventListener#matching_event_listeners_for_removal

Reply:
> It's worth noting that some browser releases have been inconsistent on this, and unless you have specific reasons otherwise, it's probably wise to use the same values used for the call to addEventListener() when calling removeEventListener().

It's just one line away.

Reply:
There's no such parameter existed in `options` for removeListener, TS will also reject that. I've tried while pulling this PR, that's why I didn't add it

Reply:
React also ignores `passive` option. So if it leaks in an obscure browser, the leaks from React will be worse.